### PR TITLE
[MDB ignore] The Corpse update

### PR DESCRIPTION
**Pull request by PalJohnson**
## About The Pull Request

image

- In short, this update seeks to bring up to the standard the various corpses around the maps of the cm13 universe. All of the corpses featured sprite work from the old ss13. This seeks to rectify it. Most of the commonly seen corpses in use for the maps have had a wardrobe change alongside loot changes in order to make the now deceased colonist something more than a decorative corpse. This update also comes with a plethora of new corpses to choose from which should help mappers appropriately decorate their maps.

## Whats New?

image

- Corpses received wardrobe update up to standard with current cm13 sprite work
- All corpse spawners have been appropriately updated due to previous inconstancy
- New corpses have been added for hostile factions like UPP and CLF. (They don't spawn with weapons as this is left for mappers to decide)
- New Corpses have been added for WY and Freelancer Factions
- New Civilian Corpses have been added
- All previous colonist corpses now have an appropriate randomizer so the corpse spawns with a different outfit every time
image
- Corporate Liasion, Scientist, and Medical Corpses now spawn with loot similarly to their Survivor counterparts
- Removed survivor pouches from corpses
image
- Corpses now spawn with winter clothing if they are on a winter-themed map
-  Corpses now spawn with guns and randomized item
- Fix inconsistent coding in the old corpse spawner. 
-  Lawyer corpse has now been replaced with the Corporate Liaison corpse
- Updated big red PMC-related inserts to have loot appropriate for PMCs

## Why It's Good For The Game

Corpse spawners got re-enabled into the game a while ago which added a bit more life into the game. They worked in adding a bit of atmosphere but it was quickly noted that the code and sprites were outdated. I took it upon myself to update the corpses so they could better fit the theme of the maps you are playing on to help sell the vibe. The PMC corpse spawner didn't even look anything like a PMC. The security corpse used old ss13 sprites. With this update, I would hope this would give maps a breath of fresh air. Instead of just ignoring corpses you can now interact with them somewhat as it provides something of value other than visual.

## Why It's Bad For The Game

Adding new corpses with updated loadouts could slow the server down a little. No new corpses were added so things should stay the same for the most part. The addition of loot in the form of guns and other items may potentially upset the balance. Given that much of the corpses are in hive or in open areas I believe this shouldn't be a problem for a xeno melting items across the map.

## Changelog

:cl: theflyingflail
add: Added new corpses to civilian, neutral, and hostile spawn pool
add: Corpses now have loot such as weapons and randomized items
add: Added new loot to PMC crashed ship insert and office insert
qol: Updated code to look less messy when dealing with corpses
qol: Updated corpse spawners on all maps and inserts
qol: Updated colonist corpses to now spawn with winter items and randomized outfits
del: Removed survivor pouch from corpses
fix: fixed inconsistent coding which led to spawning corpse errors
/:cl:

### CODE REVIEW

**Comment by MusselMain** [comment]
Cool update, liking the survivor buff

**Comment by Diegoflores31** [comment]
if the survival pouch is removed doesnt that mean it is rather a nerf ? its 1 stack of metal less per body . do they provide anything barely valuable or equivalent to that stack of metal ?

**Comment by PalJohnson** [comment]
> 

Just removed it from corpses, not survivors (living).

**Review by Agameofscones — APPROVED** [review]
mapping looks OK

**Comment by c4xmaniac** [comment]
Looks cool af, would be cool to loot dead bodies for actual loot. Gives police skill survs another looting option.

File: code/modules/gear_presets/corpses.dm [hunk 1]
```diff
@@ -471,3 +670,337 @@
...
+		H.equip_to_slot_or_del(new /obj/item/clothing/mask/rebreather/scarf/tacticalmask/green, WEAR_FACE)
+
+/datum/equipment_preset/corpse/upp/burst
+	name = "Corpse - Union of Progressive Peoples Soldier"
+	xenovictim = TRUE
+
+// PMC
+
+/datum/equipment_preset/corpse/pmc
+	name = " Corpse - PMC"
```
Comment (stanalbatross) [inline code/modules/gear_presets/corpses.dm:808]:
```suggestion
	name = "Corpse - PMC"
```

**Review by stanalbatross — CHANGES_REQUESTED** [review]
- the "burst colonist" corpses only spawn with kutjevo gear
- the "colonist random" ones spawn with nothing
- pmc corpses still spawn with survival pouches

File: code/modules/gear_presets/corpses.dm (hunk 1 above)
Comment (PalJohnson) [reply code/modules/gear_presets/corpses.dm]: Applied in the patch already. Thanks for the catch!

**Comment by stanalbatross** [comment]
image
things are getting out of hand. now there are two of them!

**Comment by PalJohnson** [comment]
> image things are getting out of hand. now there are two of them!

image
pushed patch

File: code/modules/gear_presets/corpses.dm [hunk 2]
```diff
@@ -471,3 +663,327 @@
...
+	H.equip_to_slot_or_del(new /obj/item/clothing/shoes/veteran/PMC/knife, WEAR_FEET)
+	H.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/PMC, WEAR_FACE)
+	H.equip_to_slot_or_del(new /obj/item/storage/backpack/satchel, WEAR_BACK)
+	H.equip_to_slot_or_del(new /obj/item/device/radio(H), WEAR_IN_BACK)
+	H.equip_to_slot_or_del(new /obj/item/storage/belt/gun/m4a3/mod88, WEAR_WAIST)
+	H.equip_to_slot_or_del(new /obj/item/storage/pouch/tools/full(H), WEAR_R_STORE)
+	add_random_survivor_equipment(H)
+
+/datum/equipment_preset/corpse/pmc/burst
+	name = "Corpse - Weyland-Yutani PMC (Standard)"
```
Comment (stanalbatross) [inline code/modules/gear_presets/corpses.dm:829]:
```suggestion
	name = "Burst Corpse - Weyland-Yutani PMC (Standard)"
```

**Review by stanalbatross — APPROVED** [review]

**Review by stanalbatross — APPROVED** [review]

**Comment by stanalbatross** [comment]
if the weapons spawning with corpses make weapon acquisition too easy they can be removed later. good for now.

**Pull request merged**